src/GovUk.Education.ExploreEducationStatistics.Admin/Services/Public.Data/ProcessorClient.cs

```diff
@@ -59,4 +59,36 @@ await response.Content
         return content
             ?? throw new Exception("Could not deserialize the response from the Public Data Processor.");
     }
+
+    public async Task<Either<ActionResult, Unit>> DeleteDataSetVersion(
+        Guid dataSetVersionId, 
+        CancellationToken cancellationToken = default)
+    {
+        var response = await httpClient
+            .DeleteAsync($"api/DeleteDataSetVersion/{dataSetVersionId}", cancellationToken: cancellationToken);
+
+        if (!response.IsSuccessStatusCode)
+        {
+            switch (response.StatusCode)
+            {
+                case HttpStatusCode.BadRequest:
+                    return new BadRequestObjectResult(
+                        await response.Content
+                            .ReadFromJsonAsync<ValidationProblemViewModel>(cancellationToken: cancellationToken)
+                    );
+                case HttpStatusCode.NotFound:
+                    return new NotFoundResult();
+                default:
+                    var message = await response.Content.ReadAsStringAsync(cancellationToken);
+                    logger.LogError($"""
+                         Failed to delete data set version with status code: {response.StatusCode}. Message:
+                         {message}
+                         """);
+                    response.EnsureSuccessStatusCode();
+                    break;
+            }
```

Comment:
I did think of doing this, however I decided against it because they're distinctly different endpoint calls and have no reason to necessarily be handled the same way or have the same status codes to handle. It's just coincidence in this case that they're the same. In fact, originally, the processor didn't return 400s so this method didn't handle this case so was different to the `CreateInitialDataSetVersion` method. It then became identical when I added 400s to the processor endpoint. But like I say, this could easily change and become different again.

So, I'm wondering whether to leave them separate on purpose, even if there is some repetition?

One thing to consider is, I will be adding another method to this interface when I add the bulk delete endpoint to the processor, and it may respond with the same status codes again, and therefore look identical to these again. Which may sway me to do what you've said since at that point we'd have 3 methods that could reuse a lot of stuff.

In summary, I'm undecided because they feel like they shouldn't be made to try to share stuff, but the fact is they're pretty identical.

Reply:
Ok yep I thought this might be the case. I agree with your point about handling them both in a distinct way for now, and maybe this decision changes when the next operation gets added to it.